client/src/components/related/related-components/RelatedList.jsx

```diff
@@ -1,39 +1,50 @@
-// IMPORT LIBRARY
-import React, { useState, useEffect, useRef } from 'react';
-
-// IMPORT COMPONENTS AND STYLE
-import RelatedProduct from './RelatedProduct.jsx';
+/* eslint-disable no-unused-expressions */
+import React, { useState, useRef, useEffect } from 'react';
 import { AiOutlineLeftSquare, AiOutlineRightSquare } from 'react-icons/ai';
+import RelatedProduct from './RelatedProduct.jsx';
 import './styles/relatedList.css';
 
-export default function RelatedList({ feature, end, relatedList }) {
-
+export default function RelatedList({ feature, relatedList }) {
   const ref = useRef(null);
   const [posIndex, setPosIndex] = useState(0);
-  const [showModal, setShowModal] = useState(false);
+
+  const endOfRelatedList = relatedList.length ? relatedList.length - 5 : 0;
 
   const scrollLeft = () => {
     posIndex > 0 ? setPosIndex(posIndex - 1) : null;
     ref.current.scrollLeft -= 15 * 16;
   };
 
   const scrollRight = () => {
-    posIndex < end ? setPosIndex(posIndex + 1) : null;
+    posIndex < endOfRelatedList ? setPosIndex(posIndex + 1) : null;
     ref.current.scrollLeft += 15 * 16;
   };
 
+  useEffect(() => {
+
```

Comment:
Is this useEffect doing anything?

Reply:
Oh no I move it somewhere else, just delete it